src/firefly/js/charts/ui/CombineChart.jsx

```diff
@@ -113,62 +174,95 @@ function createTableModel(showAll, tbl_id) {
     return table;
 }
 
-const CombineChartDialog = ({onComplete}) => {
-
+const ChartSelectionTable = ({tbl_id, chartIds, selectedChartId}) => {
     const [showAll, setShowAll] = useState(false);
+    const showAllHint = 'Show all charts even ones that may not combine well';
+
+    useEffect(() => {
+        const selTbl = createTableModel(chartIds, selectedChartId ?? chartIds[0], showAll, tbl_id);
+        dispatchTableAddLocal(selTbl, undefined, false);
+    }, [showAll]);
+
+    return (
+        <Stack height={125}>
+            {/*--- temporarily removed to only combine charts with similar axes
+                <div style={{display: 'inline-flex', alignItems: 'center'}}>
+                    <label htmlFor='showAll'>Show All Charts: </label>
+                    <input type='checkbox' id='showAll' title={showAllHint} value={showAll} onClick={(e) => setShowAll(e.target?.checked)}/>
+                    <span className='CombineChart__hints'>({showAllHint})</span>
+                </div>
+                ----------*/}
+            <TablePanel {...{tbl_id, showToolbar: false, showUnits: false, showTypes: false}}/>
+        </Stack>
+    );
+};
+
 
+const CombineChartDialog = ({onComplete, chartIds, selectedChartId, allowChartSelection, deriveTraceTitle}) => {
     const tbl_id = 'combinechart-tbl-id';
     const groupKey = 'combinechart-props';
 
+    const [resolvedChartIds, setResolvedChartIds] = React.useState([]);
+    const [rejectedChartReasons, setRejectedChartReasons] = React.useState([]);
     const doCascading = useStoreConnector(()=>getFieldVal(groupKey, 'doCascading'));
+    const tblSelectedChartIds = useStoreConnector(()=>
+        getColumnValues(getSelectedDataSync(tbl_id), 'ChartId'));
 
     useEffect(() => {
-        const selTbl = createTableModel(showAll, tbl_id);
-        dispatchTableAddLocal(selTbl, undefined, false);
-    }, [showAll]);
+        if (chartIds[0] instanceof Promise) {
+            // store fulfilled values and rejected reasons in the state when available
+            Promise.allSettled(chartIds).then((results) => {
+                setResolvedChartIds(results.filter(({status})=> status==='fulfilled')
+                    .map(({value}) => value));
+                setRejectedChartReasons(results.filter(({status})=> status==='rejected')
+                    .map(({reason}) => reason));
+            });
+        } else {
+            setResolvedChartIds(chartIds);
+        }
+    }, [chartIds]);
+
+
+    const chartIdsToCombine = allowChartSelection
+        ? [selectedChartId ?? chartIds[0], ...tblSelectedChartIds] //selectedChartId isn't present in table
+        : resolvedChartIds; //TODO: run chartIds through canCombine() and show warning if not?
```

Comment:
I understand what `allowChartSelection` is for now.  Maybe it should be called `requiredChartSelection`?  Just a suggestion.

Reply:
This is actually a flag to control whether to render a table inside dialog that allows further selection among the passed chart IDs. I think I should rename it to `showChartSelectionTable` to resolve ambiguity - what do you think?